drivers/spi/spi_dw.c

```diff
@@ -554,7 +554,7 @@ const struct spi_dw_config spi_dw_config_0 = {
 	.regs = DT_INST_REG_ADDR(0),
 	.clock_frequency = INST_0_SNPS_DESIGNWARE_SPI_CLOCK_FREQ,
 	.config_func = spi_config_0_irq,
-	.op_modes = CONFIG_SPI_0_OP_MODES
+	.op_modes = SPI_CTX_RUNTIME_OP_MODE_MASTER
```

Comment:
wait, dts does not support slave SPI controllers?

Reply:
There is a dts property `spi-slave` that we don't utilize current.  All other drivers that seem to support CONFIG_SPI_SLAVE just assume ever instance can support either master or slave mode.